File: qiuzhao/bishi/8.7/test2.cpp
#include<iostream>
#include<string>
#include<vector>
using namespace std;
string coupons(string &s){
    string res="";
    vector<int> chNums(26,0);
    for(auto &ch:s){
        chNums[ch-'a']++;
    }
    for(int i=0;i<26;++i){
        if(chNums[i]>0){
            res.push_back(i+'a');
            --chNums[i];
        }
    }
    for(int i=25;i>=0;--i){
        while(chNums[i]>0){
            res.push_back(i+'a');
            chNums[i]--;
        }
    }
    return res;
}
int main(){
    string s;
    cin>>s;
    cout<<coupons(s)<<endl;
    return 0;
}

File: qiuzhao/bishi/8.7/test2.java
import java.util.*;

public class Main {
    public static String coupons(String s) {
        StringBuilder res = new StringBuilder();
        int[] chNums = new int[26];
        for (char ch : s.toCharArray()) {
            chNums[ch - 'a']++;
        }
        for (int i = 0; i < 26; ++i) {
            if (chNums[i] > 0) {
                res.append((char) (i + 'a'));
                chNums[i]--;
            }
        }
        for (int i = 25; i >= 0; --i) {
            while (chNums[i] > 0) {
                res.append((char) (i + 'a'));
                chNums[i]--;
            }
        }
        return res.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.next();
        System.out.println(coupons(s));
    }
}